### PR TITLE
(Maybe) bug fix

### PR DESCRIPTION
**Pull request by Dewdrops**
@roman 
I'm not sure whether they're bugs or intentionally coded. Please check the `bugfix` branch.

### CODE REVIEW

File: evil-visual-mark-mode.el [hunk 1]
```diff
@@ -52,8 +52,7 @@
 
 This marker will normally come from the advised evil-set-marker
 function."
-  (let* ((marker-pos (marker-position marker)))
-    (make-overlay marker-pos (+ 2 marker-pos))))
+  (make-overlay marker marker))
```
Comment (roman) [inline evil-visual-mark-mode.el:55]: Shouldn't we make the marker of the size of 2 to properly render the backtick char with the letter marker?

File: evil-visual-mark-mode.el [hunk 2]
```diff
@@ -109,9 +108,10 @@ This function is called on `evil-normal-state-entry-hook'."
 This function is called on the initialization of
 `evil-visual-mark-mode'"
   (evil-visual-mark-populate-overlay-alist)
-  (--each evil-visual-mark-overlay-alist
-    (evil-visual-mark-overlay-put (car it)
-                                  (cdr it))))
+  (when (evil-normal-state-p)
```
Comment (roman) [inline evil-visual-mark-mode.el:111]: Have you found a case where this code doesn't execute on evil-normal-state already?

File: evil-visual-mark-mode.el (hunk 1 above)
Comment (Dewdrops) [inline evil-visual-mark-mode.el:55]: I'm not familiar with how the overlay mechanism works underhood, perhaps I misunderstand something. 
The official Emacs Lisp reference says the `before-string` property won't appear in the buffer in any sense except for visual effects, so I think it doesn't really occupy any space that denoted by 'buffer position'. In my Emacs, it seems that the visual markers are rendered correctly even with zero-length overlays.

File: evil-visual-mark-mode.el (hunk 2 above)
Comment (Dewdrops) [inline evil-visual-mark-mode.el:111]: If enable evil-visual-mark-mode in evil-insert-state, the marks will be visible immediately before returning to evil-normal-state.
Thanks for the review. The commit message may be confusing, my English is poor, ;-).

**Comment by roman** [comment]
:+1:

**Pull request merged**